src/components/vm/VmInitializer.tsx

```diff
@@ -96,21 +98,21 @@ export default function VmInitializer() {
     });
   }, [near]);
 
-  const requestSignInWithWallet = useCallback(
-    (event: any) => {
-      event?.preventDefault();
-      walletModal?.show();
-      return false;
-    },
-    [walletModal],
-  );
+  const requestSignInWithWallet = useCallback(() => {
+    signInRedirect.saveCurrentUrl();
+    walletModal?.show();
+    return false;
+  }, [signInRedirect, walletModal]);
 
-  const requestSignIn = useCallback(
-    (queryParam?: string) => {
-      router.push(`/signin${queryParam}`);
-    },
-    [router],
-  );
+  const requestSignIn = useCallback(() => {
+    signInRedirect.saveCurrentUrl();
+    router.push('/signin');
+  }, [router, signInRedirect]);
+
+  const requestCreateAccount = useCallback(() => {
+    signInRedirect.saveCurrentUrl();
+    router.push('/signup');
+  }, [router, signInRedirect]);
```

Comment:
How about this
```suggestion
  const requestAuthentication = useCallback(() => (redirect) =>{
    signInRedirect.saveCurrentUrl();
    router.push(redirect);
  }, [router, signInRedirect]);
```
?
I mean this two methods are almost the same so why do we them separately?

Reply:
Good point. I refactored these two methods into a single method provided by the `useSignInRedirect()` hook. Take a look and let me know what you think.